_rules/aria-alertdialog-input-error-6f484a.md

```diff
@@ -0,0 +1,919 @@
+---
+id: 6f484a
+name: aria-alertdialog identifies input error
+rule_type: atomic
+
+description: |
+  This rule checks that `aria-alertdialog` is used to identify input errors.
+
+accessibility_requirements: # Remove whatever is not applicable
+  wcag-technique:ARIA18: # Using aria-alertdialog to Identify Errors
+    forConformance: false
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+
+authors:
+  - Carlos Duarte
+  - João Vicente
+---
+
+## Applicability
+
+The rule applies to each HTML or SVG element:
+
+- the has one of the following [semantic roles](#semantic-role): `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`;
+- that may or may not belong to a [form element](https://www.w3.org/TR/html52/sec-forms.html#the-form-element);
+- for which [input errors](https://www.w3.org/TR/WCAG21/#dfn-input-error) are [automatically detected](#automatic-error-detection).
+
+**Note**: The list of applicable [semantic roles](#semantic-role) is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
+
+- inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+## Expectation
+
+After [user completion](#completed-input-field) of the target element or triggering the submission of the form, if the target element belongs to one, each target element for which an [input error](https://www.w3.org/TR/WCAG21/#dfn-input-error) has been [automatically detected](#automatic-error-detection) causes an element with `role="alertdialog"` to appear, where the following is true:
+
+- it contains at least one [focusable](#focusable) element; and
```

Comment:
The word "contains" is ambiguous; are we looking for a [child](https://dom.spec.whatwg.org/#concept-tree-child), a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant), a [shadow-including descendant](https://dom.spec.whatwg.org/#concept-shadow-including-descendant), or something else entirely?